src/mac/streams/system_info.rs

```diff
@@ -0,0 +1,200 @@
+use super::*;
+use crate::minidump_format::*;
+
+/// Retrieve the OS version information.
+///
+/// Note that this only works on 10.13.4+, but that release is over 4 years old
+/// and 1 version behind the latest unsupported release at the time of this writing
+///
+/// Note that Breakpad/Crashpad use a private API in CoreFoundation to do this
+/// via _CFCopySystemVersionDictionary->_kCFSystemVersionProductVersionKey
+fn os_version() -> (u32, u32, u32) {
+    let vers = mach::sysctl_string(b"kern.osproductversion\0");
+
+    let inner = || {
+        let mut it = vers.split('.');
+
+        let major: u32 = it.next()?.parse().ok()?;
+        let minor: u32 = it.next()?.parse().ok()?;
+        let patch: u32 = it.next().and_then(|p| p.parse().ok()).unwrap_or_default();
+
```

Comment:
The version is _usually_ a standard semver `<major>.<minor>.<patch>` but it _can_ be just `<major>.<minor>`